File: DSA/tower_of_hanoi.c
#include <stdio.h>

int main() {
  void tower(int n, char beg, char aux, char end);
  tower(4, 'a', 'b', 'c');
  return 0;
}

void tower(int n, char beg, char aux, char end) {
  if (n <= 0)
    printf("Illegal");
  else if (n == 1)
    printf("Move disc from %c to %c\n", beg, end);
  else {
    tower(n - 1, beg, end, aux);
    tower(1, beg, aux, end);
    tower(n - 1, aux, beg, end);
  }
}

File: Advanced Competetive Coding 2/TowerOfHanoi.java
public class TowerOfHanoi {
  public static void tower(int n, char beg, char aux, char end) {
    if (n <= 0)
      System.out.println("Illegal");
    else if (n == 1)
      System.out.println("Move disc from " + beg + " to " + end);
    else {
      tower(n - 1, beg, end, aux);
      tower(1, beg, aux, end);
      tower(n - 1, aux, beg, end);
    }
  }

  public static void main(String[] args) {
    tower(3, 'S', 'A', 'D');
  }

}
